src/network/core/tcp_http.cpp

```diff
@@ -42,19 +42,16 @@ NetworkHTTPSocketHandler::NetworkHTTPSocketHandler(SOCKET s,
 	redirect_depth(depth),
 	sock(s)
 {
-	size_t bufferSize = strlen(url) + strlen(host) + strlen(GetNetworkRevisionString()) + (data == nullptr ? 0 : strlen(data)) + 128;
-	char *buffer = AllocaM(char, bufferSize);
-
 	Debug(net, 5, "[tcp/http] Requesting {}{}", host, url);
+	std::string request;
 	if (data != nullptr) {
-		seprintf(buffer, buffer + bufferSize - 1, "POST %s HTTP/1.0\r\nHost: %s\r\nUser-Agent: OpenTTD/%s\r\nContent-Type: text/plain\r\nContent-Length: %d\r\n\r\n%s\r\n", url, host, GetNetworkRevisionString(), (int)strlen(data), data);
+		request = fmt::format("POST {} HTTP/1.0\r\nHost: {}\r\nUser-Agent: OpenTTD/{}\r\nContent-Type: text/plain\r\nContent-Length: {}\r\n\r\n{}\r\n", url, host, GetNetworkRevisionString(), strlen(data), data);
 	} else {
-		seprintf(buffer, buffer + bufferSize - 1, "GET %s HTTP/1.0\r\nHost: %s\r\nUser-Agent: OpenTTD/%s\r\n\r\n", url, host, GetNetworkRevisionString());
+		request = fmt::format("GET {} HTTP/1.0\r\nHost: {}\r\nUser-Agent: OpenTTD/{}\r\n\r\n", url, host, GetNetworkRevisionString());
```

Comment:
despite the fact that this file clearly has `debug.h` included, I wonder if we should include `fmt` too, just in case someone removes `debug.h`. I often wonder this, about when we do add an include and when not. It is a bit hit and go :)

Reply:
I rather do not add more `#include`s than needed. Even though you then "obfuscate" what you're doing, but otherwise I would have to add `#include <string>` about everywhere as well.